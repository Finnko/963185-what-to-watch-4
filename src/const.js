const Config = {
  MOVIE_MAX_RATING: 10,
  MOVIE_ACTORS_TO_SHOW: 4,
  SIMILAR_MOVIES_TO_SHOW: 4,
  MOVIES_NUMBER_TO_SHOW: 16,
  MOVIES_NUMBER_LIMIT: 8,
  MOVIES_NUMBER_OFFSET: 8,
  MAX_GENRES_TO_SHOW: 9,
  MOVIE_RATING_MAP: {
    0: `Bad`,
    3: `Normal`,
    5: `Good`,
    8: `Very good`,
    10: `Awesome`
  },
  ERRORS: {
    FETCH_DATA: `Something went wrong, please try again later.`,
  },
  LOADER: {
    SMALL: `small`,
    MEDIUM: `medium`,
  },
  DEFAULT_FILTER: `All genres`,
};

const PathName = {
  ROOT: `/`,
  MOVIE_PAGE: `/movie/`,
};

const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

const ViewMode = {
  PROMO: {
    MAIN: `MAIN`,
    DETAILS: `DETAILS`
  },
  MOVIE_CARD: {
    WITH_PLAYER: `WITH_PLAYER`,
    IMAGE: `IMAGE`,
  }
};

export {
  Config,
  PathName,
  TabName,
  ViewMode,
};
