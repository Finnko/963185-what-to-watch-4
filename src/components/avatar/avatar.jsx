import React from 'react';
import PropTypes from 'prop-types';
import UserPropType from '../../prop-types/user';
import {AuthStatus, Config} from '../../const';
import {connect} from 'react-redux';
import {getAuthStatus, getUser} from '../../store/reducers/user/selectors';

const Avatar = ({userInfo, authStatus}) => {
  let avatar = ``;
  if (userInfo !== null) {
    avatar = userInfo.avatarUrl;
  }

  const isLogged = AuthStatus.AUTH === authStatus;

  return (
    <div className="user-block">
      <div className="user-block__avatar">
        <img
          src={`${isLogged ? `${Config.SERVER_ORIGIN}${avatar}` : `img/avatar.jpg`}`}
          alt="User avatar"
          width="63"
          height="63"
        />
      </div>
    </div>
  );
};

Avatar.propTypes = {
  authStatus: PropTypes.string,
  userInfo: UserPropType,
};

const mapStateToProps = (state) => {
  return {
    authStatus: getAuthStatus(state),
    userInfo: getUser(state),
  };
};

export {Avatar};
export default connect(mapStateToProps)(Avatar);
