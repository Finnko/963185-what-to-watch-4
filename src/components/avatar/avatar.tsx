import * as React from 'react';
import {AuthStatus, Config, PathName} from '../../const';
import {connect} from 'react-redux';
import {getAuthStatus, getUser} from '../../store/reducers/user/selectors';
import {Link} from 'react-router-dom';
import {User} from "../../interfaces";

type AvatarProps = {
  userInfo: User;
  authStatus: string;
}

const Avatar: React.FC<AvatarProps> = (props: AvatarProps) => {
  const {userInfo, authStatus} = props;

  let avatar = ``;
  if (userInfo !== null) {
    avatar = userInfo.avatarUrl;
  }

  const isLogged = AuthStatus.AUTH === authStatus;

  return (
    <div className="user-block">
      {isLogged
        ?
        <Link
          to={PathName.MY_LIST}
          className="user-block__avatar"
        >
          <img
            src={`${isLogged ? `${Config.SERVER_ORIGIN}${avatar}` : `img/avatar.jpg`}`}
            alt="User avatar"
            width="63"
            height="63"
          />
        </Link>
        :
        <Link
          to={PathName.SIGN_IN}
          className="user-block__link"
        >
            Sign In
        </Link>
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    authStatus: getAuthStatus(state),
    userInfo: getUser(state),
  };
};

export {Avatar};
export default connect(mapStateToProps)(Avatar);
