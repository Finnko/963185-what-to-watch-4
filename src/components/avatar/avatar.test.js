import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import NameSpace from '../../store/name-space';
import {Provider} from 'react-redux';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom';
import {Avatar} from './avatar.jsx';
import {AuthStatus} from '../../const';

const mockUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const props = {
  userInfo: mockUser,
};

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.USER]: {
    user: mockUser,
    authStatus: AuthStatus.NO_AUTH,
  }
});

describe(`Avatar component render correctly`, () => {
  it(`Should Avatar component render correctly when user not logged`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <Avatar authStatus={AuthStatus.NO_AUTH} {...props}/>
            </Router>
          </Provider>

      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Avatar component render correctly when user was authorized`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <Avatar authStatus={AuthStatus.AUTH} {...props}/>
            </Router>
          </Provider>

      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
