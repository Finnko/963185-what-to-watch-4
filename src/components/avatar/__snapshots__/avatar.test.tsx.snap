// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component render correctly Should Avatar component render correctly when user not logged 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign In
  </a>
</div>
`;

exports[`Avatar component render correctly Should Avatar component render correctly when user was authorized 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/mylist"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="https://4.react.pages.academyimg/1.png"
      width="63"
    />
  </a>
</div>
`;
